cmd/api/src/api/v2/database_management.go

```diff
@@ -0,0 +1,155 @@
+		nodeIDs []graph.ID
+		options []string
+	)
+
+	if err := api.ReadJSONRequestPayloadLimited(&payload, request); err != nil {
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusBadRequest, "JSON malformed.", request),
+			response,
+		)
+	}
+
+	// delete graph
+	if payload.CollectedGraphData {
+		options = append(options, "collected graph data")
+
+		if err := s.Graph.ReadTransaction(request.Context(), func(tx graph.Transaction) error {
+			fetchedNodeIDs, err := ops.FetchNodeIDs(tx.Nodes())
+
+			nodeIDs = append(nodeIDs, fetchedNodeIDs...)
+			return err
+		}); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "error fetching all nodes", err.Error()), request),
+				response,
+			)
+			return
+		}
+
+		if err := s.Graph.BatchOperation(request.Context(), func(batch graph.Batch) error {
+			for _, nodeId := range nodeIDs {
+				if err := batch.DeleteNode(nodeId); err != nil {
+					return err
+				}
+			}
+			return nil
+		}); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "error deleting all nodes", err.Error()), request),
+				response,
+			)
+			return
+		}
+
+	}
+
+	// delete custom high value selectors
+	if payload.HighValueSelectors {
+		options = append(options, "custom high value selectors")
+
+		if payload.AssetGroupId == 0 {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusBadRequest, "please provide an assetGroupId to delete", request),
+				response,
+			)
+			return
+		}
+		if err := s.DB.DeleteAssetGroupSelectors(request.Context(), payload.AssetGroupId); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s %d: %s", "there was an error deleting asset group with id = ", payload.AssetGroupId, err.Error()), request),
+				response,
+			)
+			return
+		}
+	}
+
+	// delete file ingest history
+	if payload.FileIngestHistory {
+		options = append(options, "file ingest history")
+
+		if err := s.DB.DeleteAllFileUploads(); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "there was an error deleting file ingest history", err.Error()), request),
+				response,
+			)
+			return
+		}
+	}
+
+	// delete data quality history
+	if payload.DataQualityHistory {
+		options = append(options, "data quality history")
+
+		if err := s.DB.DeleteAllDataQuality(); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "there was an error deleting data quality history", err.Error()), request),
+				response,
+			)
+			return
+		}
+	}
+
+	if err := s.DB.AppendAuditLog(request.Context(), model.AuditEntry{
```

Comment:
Flagging @juggernot325 since he is up-to-date with all things audit logging. Can you take a look here and validate that this is the appropriate way to create audit logs in this scenario?

Reply:
I know we discussed this while you were working on it and decided on generating the audit logs in this method. After looking at the PR as a whole I think we may want to go a different route. One thing that I think is big concern is that if there are any errors in the above operations there will not be an audit log generated at all, so that should be addressed.

I actually think this is going to need a novel implementation of audit logging given the complexity of this operation. Let's meet up and discuss some options.

Reply:
The latest iteration of this looks good to me! It should successfully capture the intent audit log and capture actual results.